deploy/docker/docker-entrypoint.sh

```diff
@@ -116,50 +116,25 @@ echo_value() {
 # fill config on specific file if the key exists
 # SYNOPSIS
 #     try_fill_config FILE KEY VALUE
-try_fill_config() {
-    local file=$1
-    local key=$2
-    local value=$3
-    local escaped_key
-    # shellcheck disable=SC2001
-    escaped_key=$(echo "$key" | sed 's/[^a-zA-Z0-9_]/\\&/g')
-    local escaped_value
-    escaped_value=$(echo "$value" | sed 's/[\/&]/\\&/g')
-    if grep -qE "^[#[:space:]]*$escaped_key\s*=" "$file"; then
-        echo_value "$key" "$value"
-        if [[ -z "$value" ]]; then
-            sed -i -r "s/^[#[:space:]]*($escaped_key)\s*=\s*(.*)/# \1 = \2/" "$file"
-        else
-            sed -i -r "s/^[#[:space:]]*($escaped_key)\s*=\s*(.*)/\1 = $escaped_value/" "$file"
-        fi
-    # Check if config has a numbering system, but no existing configuration line in file
-    elif echo "$key" | grep -qE '\.\d+|\d+\.'; then
-        if [[ -n "$value" ]]; then
-            local template
-            template="$(echo "$escaped_key" | sed -r -e 's/\\\.[0-9]+/\\.[0-9]+/g' -e 's/[0-9]+\\\./[0-9]+\\./g')"
-            if grep -qE "^[#[:space:]]*$template\s*=" "$file"; then
-                echo_value "$key" "$value"
-                sed -i '$a'\\ "$file"
-                echo "$key = $value" >> "$file"
-            fi
-        fi
+override_config() {
+    local key=$1
+    local value=$2
+    local conf
+    conf=$(/usr/bin/lookup-plugin.sh "$key" "$_EMQX_HOME")
+    if [[ -z "$value" ]] || [[ "$value" = "null" ]]; then
+      echo "$key = null" >> "$conf.override"
+    else
+      echo "$key = $value" >> "$conf.override"
```

Comment:
if we quote here, users should not have to escape-quote in bash.
i.e.
`key = "\"value\"" >> "$conf.override"
exception will have to be made for numbers.

Reply:
if we do so, we will not be able to distinguish between "true" and true. ditto for false and null

Reply:
But if you are ok with that, I agree with your proposal because it looks more helpful.

Reply:
yeah, there isn't many different types in hocon.

- so, for `null` we already have a check,
- we need to test if it's `true|false`
- we need to test if it's a number (including float point numbers)
- then everything else can be quoted.

```
if [[ "$value"  == "" ]] || [[ "$value" == "null" ]]; then
  # do nothing
elif [[ "$value" == "true" ]] || [[ "$value" == "false" ]]; then
  # do nothing
elif  [[ $value =~ ^[+-]?[0-9]+([.][0-9]+)?$ ]] ; then
  # do nothing
else
  value="\"$value\""
fi
```